compiler/packages/babel-plugin-react-compiler/src/HIR/BuildHIR.ts

```diff
@@ -2427,6 +2427,21 @@ function lowerExpression(
       let expr = exprPath as NodePath<t.TSNonNullExpression>;
       return lowerExpression(builder, expr.get("expression"));
     }
+    case "MetaProperty": {
+      let expr = exprPath as NodePath<t.MetaProperty>;
+      if (
+        expr.node.meta.name === "import" &&
+        expr.node.property.name === "meta"
+      ) {
+        return {
+          kind: "MetaProperty",
+          meta: expr.node.meta.name,
+          property: expr.node.property.name,
+          loc: expr.node.loc ?? GeneratedSource,
+        };
+      }
+      return { kind: "UnsupportedNode", node: exprNode, loc: exprLoc };
```

Comment:
Note that it’s not enough to return an unsupported node, we also have to actually record a todo error for this case. 

Please also add a fixture using `new.target` to demonstrate that we reject it. You can name the fixture with “error.todo-new-target-meta-property.js” - the “error” prefix tells our test runner to fail unless the fixture causes an error. And the “todo-“ is a reminder to us that the input should someday be supported.